app/models/application_record.rb

```diff
@@ -19,8 +22,13 @@ def log_errors
   # Cached for performance.
   def mass_validation_enabled?
     if AppConfig.table_exists? # for migrations previous to AppConfig creation
-      return Rails.cache.fetch("ApplicationRecord.mass_validation_enabled?") do
-        AppConfigHelper.get_app_config(AppConfig::ENABLE_MASS_VALIDATION)
+      if !@@mass_validation_enabled.nil?
```

Comment:
I really don't think we need cache at all. 

And here `@@mass_validation_enabled ||= AppConfigHelper.get_app_config(AppConfig::ENABLE_MASS_VALIDATION)` should be enough...

@davidrissato Why would you say we need redis cache?

Reply:
We actually don’t need redis cache, a constant is enough. We just need to make sure this line you suggested produces the effect we want, because that is an abstract class, and I believe the behavior of @@ is different in this case.

Reply:
I updated to use a class var only, and I retested that it is only set once.

> I believe the behavior of @@ is different in this case

I couldn't find any references to that. Any pointers? My understanding of ruby class vars is that they act as global vars.